### PR TITLE
Feature/enable caching

### PR DESCRIPTION
**Pull request by tofixx**

### CODE REVIEW

File: middleware/imagePersistance.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+const hash = require('object-hash');
+const fileCache = require('../middleware/fileCache');
+
+function imagePersistance(req, res, next) {
```
Comment (franticBIT) [inline middleware/imagePersistance.js]: imagePersist**e**nce 😜

File: middleware/checkCache.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
+const cache = require('../middleware/fileCache');
+
+function checkCache(req, res, next) {
+  let etag = req.get('If-None-Match');
+  console.log('etag received', etag);
+
+  if (etag && cache.exists(etag)) {
+    // respond with 403 if file exist and has not been changed
+    res.set('Etag', etag);
+    res.status(304).end();
```
Comment (franticBIT) [inline middleware/checkCache.js]: I think the function should return here.

File: controllers/convert.js [hunk 3]
```diff
@@ -29,9 +29,12 @@ function convert(req, res, next) {
       .toFormat(format, options);
     res.type(format.id);
   } else {
-    res.type(req.imageProperties.mime);
+    res.type(format.id);
```
Comment (franticBIT) [inline controllers/convert.js]: Looks like this can be done outside the if-else construct, because we set it anyway to `format.id`.

**Comment by franticBIT** [comment]
Does this close #19?

File: middleware/checkCache.js [hunk 4]
```diff
@@ -0,0 +1,23 @@
+const cache = require('../middleware/fileCache');
+const hash = require('object-hash');
+
+function checkCache(req, res, next) {
+  // generate hash and check if image exists
+  // FIXME: req.accepts('image/webp') does not work
```
Comment (maxklenk) [inline middleware/checkCache.js]: check does not work, try it in safari

File: config/default.json [hunk 5]
```diff
@@ -20,5 +20,9 @@
         "DefaultGravity": "entropy",
         "Progressive": true,
         "RemoveMetadata": true
+    },
+    "Caching":{
+        "Expires": 2629000000, // one month
```
Comment (tofixx) [inline config/default.json]: it seems... you don't like red...

**Pull request merged**